src/main/java/net/neoforged/moddevgradle/internal/RunUtils.java

```diff
@@ -219,6 +222,16 @@ public static Project findSourceSetProject(Project someProject, SourceSet source
         throw new IllegalArgumentException("Could not find project for source set " + someProject);
     }
 
+    public static Map<String, String> replaceModClassesEnv(RunModel model, Supplier<ModFoldersProvider> newProvider) {
```

Comment:
`Supplier<ModFoldersProvider>` is essentially `Supplier<something lazy already>`, why is the outer supplier necessary?